articles/quickstart/webapp/nodejs/01-login.md

```diff
@@ -14,4 +16,239 @@ github:
+
+```js
+// app.js
+
+var session = require('express-session');
+
+//session-related stuff
+var sess = {
+ secret: 'CHANGE THIS SECRET',
+ cookie: {},
+ resave: false,
+ saveUninitialized: true
+};
+
+if (app.get('env') === 'production') {
+ sess.cookie.secure = true; // serve secure cookies, requires https
+}
+
+app.use(session(sess));
+```
+
+### Configure Passport with the Auth0 Strategy and Your Auth0 Application Settings
+
+In `app.js`, include the `passport` and `passport0-auth0`modules, configure Passport to use a new instance of `Auth0Strategy` with your Auth0 application settings. Load the `passport.initialize()` and `passport.session()` middlewares to initialize Passport with persistent login sessions. 
+
+Please use the client secret from the application settings in your Auth0 dashboard.
+
+```js
+// app.js
+var Auth0Strategy = require('passport-auth0'),
+    passport = require('passport');
+
+//passport-auth0
+var strategy = new Auth0Strategy({
+  domain: '${account.namespace}',
+  clientID: '${account.clientId}',
+  clientSecret: 'YOUR_CLIENT_SECRET',
+  callbackURL: 'http://localhost:3000/callback'
+ },
+ function(accessToken, refreshToken, extraParams, profile, done) {
+   // accessToken is the token to call Auth0 API (not needed in the most cases)
+   // extraParams.id_token has the JSON Web Token
+   // profile has all the information from the user
+   return done(null, profile);
+ }
+);
+
+passport.use(strategy);
+
+app.use(passport.initialize());
+app.use(passport.session());
+```
+
+### Storing and Retrieving User Data from the Session
+
+In a typical web application, the credentials used to authenticate a user will only be transmitted during the login request. If authentication succeeds, a session will be established and maintained via a cookie set in the user's browser. Each subsequent request will not contain credentials, but rather the unique cookie that identifies the session.
+
+In order to support login sessions, Passport will serialize and deserialize user instances to and from the session. Optionally, you may want to serialize only a subset to reduce the footprint, i.e., `user.id`.
+
+```js
+//app.js
+passport.serializeUser(function(user, done) {
+  done(null, user);
+});
+
+passport.deserializeUser(function(user, done) {
+  done(null, user);
+});
+```
+
+## Implementing Login, User Profile and Logout
+
+### Adding the Routes
+
+In this example, we will implement the following routes:
+
+* a route `/login`, that will trigger authentication by calling Passport's `authenticate` method. The user will be redirected to the login page.
+* a route `/callback`, that will be finalize the authentication when the user returns from the Auth0 login page and redirects to the profile page on sucessful login.
+* a `/user` route, to display the user's profile.
```

Comment:
>"that be finalize",

"that will be called when authentication finishes". 

Is not that the user returns, he'l be returned from Auth0